React/Modules/RCTUIManager.m

```diff
@@ -224,6 +224,8 @@ @implementation RCTUIManager
   NSDictionary *_componentDataByName;
 
   NSMutableSet<id<RCTComponent>> *_bridgeTransactionListeners;
+  NSMutableSet<id<RCTUIManagerObserver>> *_uiManagerObservers;
```

Comment:
Also trying to get the array to behave well might not be a good use of time now, you could change the property to be just `__weak id<RCTUIManagerObserver> _observer` similar to a traditional delegate.